diff --git a/kcms/about-distro/src/opengl-helper/main.cpp b/kcms/about-distro/src/opengl-helper/main.cpp
--- a/kcms/about-distro/src/opengl-helper/main.cpp
+++ b/kcms/about-distro/src/opengl-helper/main.cpp
@@ -10,7 +10,12 @@
 #include <QOffscreenSurface>
 #include <QOpenGLContext>
 #include <QOpenGLFunctions>
+#if __has_include(<QVulkanInstance>)
 #include <QVulkanInstance>
+#define KINFOCENTER_VK_TYPE_OTHER VK_PHYSICAL_DEVICE_TYPE_OTHER
+#else
+#define KINFOCENTER_VK_TYPE_OTHER 0
+#endif
 
 using namespace Qt::StringLiterals;
 
@@ -55,7 +60,7 @@
 
     const QJsonArray devicesArray{QJsonObject{
         {u"name"_s, glDeviceName()},
-        {u"vkDeviceType"_s, VK_PHYSICAL_DEVICE_TYPE_OTHER},
+        {u"vkDeviceType"_s, KINFOCENTER_VK_TYPE_OTHER},
     }};
 
     std::cout << QJsonDocument(devicesArray).toJson(QJsonDocument::Compact).toStdString() << '\n';
