--- a/src/helpers/killer/killer.cpp
+++ b/src/helpers/killer/killer.cpp
@@ -31,8 +31,10 @@
 #include <qpa/qplatformwindow_p.h>
 
+#if defined(Q_OS_LINUX) && !defined(MESA_EGL_NO_X11_HEADERS)
 #include <private/qtx11extras_p.h>
 #include <xcb/xcb.h>
+#endif
 
 #include <cerrno>
 #include <csignal>
@@ -160,12 +162,14 @@
         return EXIT_FAILURE;
     }
 
+#if defined(Q_OS_LINUX) && !defined(MESA_EGL_NO_X11_HEADERS)
     xcb_connection_t *connection = xcb_connect(display_string.toLatin1(), nullptr);
     if (xcb_connection_has_error(connection)) {
         std::cerr << "xcb connection failed" << std::endl;
         return EXIT_FAILURE;
     }
     xcb_timestamp_t timestamp = parser.value(timestampOption).toULong(&time_ok);
+#endif
 
     return EXIT_FAILURE;
 }
