--- a/src/core/renderdevice.cpp
+++ b/src/core/renderdevice.cpp
@@ -53,16 +53,16 @@
-    auto [result, instance] = context.createInstance(instanceInfo);
-    if (result != vk::Result::eSuccess && !validationLayers.empty()) {
+    auto expectedInstance = context.createInstance(instanceInfo);
+    if (expectedInstance.error() != vk::Result::eSuccess && !validationLayers.empty()) {
         // try again without the validation layer
         validationLayers.clear();
         instanceInfo.setPEnabledLayerNames(validationLayers);
-        auto [result, instance] = context.createInstance(instanceInfo);
-        if (result == vk::Result::eSuccess) {
+        expectedInstance = context.createInstance(instanceInfo);
+        if (expectedInstance.error() == vk::Result::eSuccess) {
             qCWarning(KWIN_CORE, "Vulkan validation layer is not installed");
-            return std::move(instance);
+            return std::move(expectedInstance.value());
         }
     }
-    return std::move(instance);
+    return std::move(expectedInstance.value());
 }
--- a/src/vulkan/vulkan_device.cpp
+++ b/src/vulkan/vulkan_device.cpp
@@ -70,1 +70,1 @@
-    auto [result, cmdPool] = m_logical.createCommandPool(vk::CommandPoolCreateInfo{
+    auto expectedCmdPool = m_logical.createCommandPool(vk::CommandPoolCreateInfo{
@@ -72,3 +72,3 @@
-    if (result != vk::Result::eSuccess) {
-        qCCritical(KWIN_VULKAN) << "creating a command pool failed:" << vk::to_string(result);
+    if (expectedCmdPool.error() != vk::Result::eSuccess) {
+        qCCritical(KWIN_VULKAN) << "creating a command pool failed:" << vk::to_string(expectedCmdPool.error());
         return;
     }
-    m_commandPool = std::move(cmdPool);
+    m_commandPool = std::move(expectedCmdPool.value());
