diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1c8fa28..6fa4afb 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,3 @@
-add_subdirectory(backendlauncher)
-add_subdirectory(doctor)
-add_subdirectory(libdpms)
 set(libkscreen_SRCS
     abstractbackend.cpp abstractbackend.h
     backendmanager.cpp
diff --git a/src/backendmanager.cpp b/src/backendmanager.cpp
index c674f4a..0428b4c 100644
--- a/src/backendmanager.cpp
+++ b/src/backendmanager.cpp
@@ -24,7 +24,6 @@
 #include <QGuiApplication>
 #include <QStandardPaths>
 #include <QThread>
-#include <QtGui/private/qtx11extras_p.h>
 
 #include <memory>
 
@@ -130,7 +129,7 @@ QFileInfo BackendManager::preferredBackend(const QString &backend)
     } else if (!env_kscreen_backend.isEmpty()) {
         backendFilter = env_kscreen_backend;
     } else {
-        if (QX11Info::isPlatformX11()) {
+        if (QGuiApplication::platformName() == QLatin1String("xcb")) {
             backendFilter = QStringLiteral("XRandR");
         } else if (QGuiApplication::platformName().startsWith(QLatin1String("wayland"))) {
             backendFilter = QStringLiteral("KWayland");
