diff --git a/src/libdpms/dpms.cpp b/src/libdpms/dpms.cpp
index e2ea1c8..e863062 100644
--- a/src/libdpms/dpms.cpp
+++ b/src/libdpms/dpms.cpp
@@ -5,17 +5,13 @@
 #include "dpms.h"
 #include "kscreendpms_debug.h"
 #include "waylanddpmshelper_p.h"
-#include "xcbdpmshelper_p.h"
 
 #include <QGuiApplication>
-#include <QtGui/private/qtx11extras_p.h>
 
 KScreen::Dpms::Dpms(QObject *parent)
     : QObject(parent)
 {
-    if (QX11Info::isPlatformX11()) {
-        m_helper.reset(new XcbDpmsHelper);
-    } else if (QGuiApplication::platformName().startsWith(QLatin1String("wayland"), Qt::CaseInsensitive)) {
+    if (QGuiApplication::platformName().startsWith(QLatin1String("wayland"), Qt::CaseInsensitive)) {
         m_helper.reset(new WaylandDpmsHelper);
     } else {
         qCWarning(KSCREEN_DPMS) << "Platform is not Wayland or X11, this doesn't make sense. Platform name is" << QGuiApplication::platformName();
