diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1c8fa28..6fa4afb 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,5 @@
 add_subdirectory(backendlauncher)
 add_subdirectory(doctor)
-add_subdirectory(libdpms)
 set(libkscreen_SRCS
     abstractbackend.cpp abstractbackend.h
     backendmanager.cpp
diff --git a/src/backendmanager.cpp b/src/backendmanager.cpp
index c674f4a..0428b4c 100644
--- a/src/backendmanager.cpp
+++ b/src/backendmanager.cpp
@@ -24,7 +24,6 @@
 #include <QGuiApplication>
 #include <QStandardPaths>
 #include <QThread>
-#include <QtGui/private/qtx11extras_p.h>
 
 #include <memory>
 
@@ -130,7 +129,7 @@ QFileInfo BackendManager::preferredBackend(const QString &backend)
     } else if (!env_kscreen_backend.isEmpty()) {
         backendFilter = env_kscreen_backend;
     } else {
-        if (QX11Info::isPlatformX11()) {
+        if (QGuiApplication::platformName() == QLatin1String("xcb")) {
             backendFilter = QStringLiteral("XRandR");
         } else if (QGuiApplication::platformName().startsWith(QLatin1String("wayland"))) {
             backendFilter = QStringLiteral("KWayland");
diff --git a/src/doctor/CMakeLists.txt b/src/doctor/CMakeLists.txt
index aefb6a2..a71e0ff 100644
--- a/src/doctor/CMakeLists.txt
+++ b/src/doctor/CMakeLists.txt
@@ -1,6 +1,6 @@
 add_executable(kscreen-doctor main.cpp doctor.cpp doctor.h)
 
-target_link_libraries(kscreen-doctor Qt::DBus KF6::Screen KF6::ScreenDpms)
+target_link_libraries(kscreen-doctor Qt::DBus KF6::Screen)
 install(TARGETS kscreen-doctor ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
 install(FILES kscreen-doctor.zsh RENAME _kscreen-doctor DESTINATION ${KDE_INSTALL_ZSHAUTOCOMPLETEDIR})
